pkg/labels/fake_applicator.go

```diff
@@ -117,6 +118,79 @@ func (app *fakeApplicator) WatchMatches(selector labels.Selector, labelType Type
 	return ch
 }
 
+func (app *fakeApplicator) WatchDiff(labelType Type, quitCh <-chan struct{}) <-chan *LabeledChanges {
```

Comment:
Why is this necessary? Shouldn't watching a daemon set's node selector be sufficient for detecting node changes?